commit.go

```diff
@@ -371,32 +369,33 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
 		}
 	}
 
-	img, err := is.Transport.GetStoreImage(b.store, dest)
-	if err != nil && !errors.Is(err, storage.ErrImageUnknown) {
-		return imgID, nil, "", fmt.Errorf("locating image %q in local storage: %w", transports.ImageName(dest), err)
-	}
-	if err == nil {
-		imgID = img.ID
-		toPruneNames := make([]string, 0, len(img.Names))
-		for _, name := range img.Names {
-			if nameToRemove != "" && strings.Contains(name, nameToRemove) {
-				toPruneNames = append(toPruneNames, name)
+	dref := dest.DockerReference()
+	if dref != nil {
+		// Check if reference is on the storage
+		if _, err := b.store.Image(dref.String()); err == nil {
+			// resolve reference
+			dest2, img, err := is.ResolveReference(dest)
+			if err != nil {
```

Comment:
This doesn’t work with ID-only references (or correctly look up name@id references); and it still touches c/storage for non-c/storage destinations like a direct commit to `docker://`.

```go
if dest.Transport().Name() == is.Transport.Name() {
   dest2, img, err := is.ResolveReference(dest)
   if err != nil { /* fail, “unknown image” is not really an expected error after we hav just created it */ }
   dest = dest2 // immediately, to make sure all following code uses exactly the same image
}
```
similarly to the check around `AdditionalTags`.

Reply:
Done.